test/test_charm.py

```diff
@@ -560,27 +560,42 @@ def _on_foo_bar_action(self, event: ops.ActionEvent):
         charm.on.foo_bar_action.emit(id='1')
 
 
-def test_action_event_defer_fails(
-    request: pytest.FixtureRequest, monkeypatch: pytest.MonkeyPatch, fake_script: FakeScript
+@pytest.mark.parametrize(
+    'event,kwargs',
+    [
+        ('start_action', {'id': 2}),
+        ('stop', {}),
+        ('remove', {}),
+        ('secret_expired', {'id': 'secret:123', 'label': None, 'revision': 0}),
+        ('secret_rotate', {'id': 'secret:234', 'label': 'my-secret'}),
+    ],
+)
+def test_inappropriate_event_defer_fails(
+    event: str,
+    kwargs: typing.Dict[str, typing.Any],
+    request: pytest.FixtureRequest,
+    monkeypatch: pytest.MonkeyPatch,
+    fake_script: FakeScript,
 ):
-    cmd_type = 'action'
-
     class MyCharm(ops.CharmBase):
         def __init__(self, framework: ops.Framework):
             super().__init__(framework)
-            framework.observe(self.on.start_action, self._on_start_action)
+            framework.observe(getattr(self.on, event), self._call_defer)
```

Comment:
I wonder if there's a better way to write this.

Reply:
Better in what way?

Reply:
Sorry, that was meant as a lead into the next comment. Closing this one.